variables/builtin.go

```diff
@@ -21,23 +21,54 @@ type DefaultArgs struct {
 	EarthlyBuildSha string
 }
 
+const (
+	VarNameEarthlyTarget             = "EARTHLY_TARGET"
+	VarNameEarthlyTargetProject      = "EARTHLY_TARGET_PROJECT"
+	VarNameEarthlyTargetProjectNoTag = "EARTHLY_TARGET_PROJECT_NO_TAG"
+	VarNameEarthlyTargetName         = "EARTHLY_TARGET_NAME"
+	VarNameEarthlyTargetTag          = "EARTHLY_TARGET_TAG"
+	VarNameEarthlyTargetTagDocker    = "EARTHLY_TARGET_TAG_DOCKER"
+	VarNameEarthlyVersion            = "EARTHLY_VERSION"
+	VarNameEarthlyBuildSha           = "EARTHLY_BUILD_SHA"
```

Comment:
Note that there are also the target platform args and the user platform args.

Reply:
There's a complete set here: https://github.com/earthly/earthly/blob/main/states/dedup/targetinput.go#L144

I wonder if we should unify the definition here with the set in `targetinput.go`. Ideally, they should be defined in a single common location. It might have to be a new package though - IIRC there was some circular dependency issue.

Reply:
Thanks for pointing that out. I see `variables/builtin.go` imports `dedup`.

I can try creating a new package such as `variables/reserved`, and move list all the strings there as consts.

Reply:
Makes sense!